### PR TITLE
feat: Cloud Run Jobs support

### LINKED ISSUE #1514: Cloud Run Jobs produce logs with incorrect resource type (not supported?)
#### Problem

Cloud Run Jobs environment and resource type detection is not supported by `MonitoredResourceUtil` and `MetadataLoader`

#### Steps to reproduce

Use the library's default configuration with Cloud Run Jobs.

#### Expected result

System out produces log entries with the following resource section:
```
resource: {
   labels: {
      job_name: "migrations"
      location: "us-central1"
      project_id: "prj-my-project"
   }
   type: "cloud_run_job"
}
```

#### Actual result

... whereas logs produced by the library have the following:
```
resource: {
   labels: {
      project_id: "prj-my-project"
   }
   type: "global"
}
```
what makes them invisible from Cloud Run Job's execution Logs UI, and you can only see them if you go to log explorer and filter out by "global" resource type.

### PR DESCRIPTION
**Pull request by kberezin-nshl**
- [x] Make sure to open an issue as a [bug](https://github.com/googleapis/java-logging/issues/1514)
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #1514 ☕️

### CODE REVIEW

**Comment by cindy-peng** [comment]
Hi @kberezin-nshl , Thank you for submitting the PR! Could you try run `mvn com.coveo:fmt-maven-plugin:format` to resolve the test failures? https://github.com/googleapis/java-logging/actions/runs/7638351592/job/20875391081?pr=1517

**Comment by kberezin-nshl** [comment]
Ah, sorry for that. 
Done, could you please try again?

**Comment by kberezin-nshl** [comment]
@cindy-peng is this going to make it into the next release? We currently have to use quite a hacky solution to make this work locally, and it would be nice to finally get this functionally normally with the next release.

**Comment by kberezin-nshl** [comment]
@cindy-peng @alicejli can we get some traction with this?

**Comment by kberezin-nshl** [comment]
@cindy-peng @alicejli  the lack of communication is quite confusing and frustrating to be honest. Is there anything we can do here? This is a massive blocker for us as we have to hack our way every time we bump the GCP dependencies versions in our project.

**Comment by cindy-peng** [comment]
> @cindy-peng @alicejli the lack of communication is quite confusing and frustrating to be honest. Is there anything we can do here? This is a massive blocker for us as we have to hack our way through every time we bump the GCP dependencies versions in our project.

Hi @kberezin-nshl, I have been out and working on other stuff. Sorry about the confusion. I will take another look and get back to you.

File: google-cloud-logging/src/main/java/com/google/cloud/logging/MetadataLoader.java [hunk 1]
```diff
@@ -182,6 +185,18 @@ private String getRegion() {
     return null;
   }
 
+  private String getCloudRunJobName() {
```
[the remark this answers is not in the document]
Comment (kberezin-nshl) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/MetadataLoader.java]: Of course. Done

**Review by cindy-peng — APPROVED** [review]

**Comment by cindy-peng** [comment]
PR approved. @kberezin-nshl In order to pass all the presubmit tests, could you try rebase the branch? We added a new test configuration previously and I didn't have the access to contribute to this PR.

**Comment by kberezin-nshl** [comment]
@cindy-peng ah, of course, done. 
I should've done that yesterday though... Anyway, hopefully all the tests will pass and this will make it to the release 🤞

**Comment by cindy-peng** [comment]
> @cindy-peng ah, of course, done. I should've done that yesterday though... Anyway, hopefully all the tests will pass and this will make it to the release 🤞

There is one dependency check failure I believe it is a known issue from our side. We can release the change once we have patched that check failure.

**Comment by cindy-peng** [comment]
Waiting for the last round of tests passing. Once that is done, we should be able to merge it and include it to the next release.

**Comment by kberezin-nshl** [comment]
> Do you mind adding me as co-owner of this PR so I can play around and check if this is anything missing?
I don't mind adding you as co-owner but I am not sure how to do that :)
I rebased and force-pushed the branch onto the latest main though.

**Comment by kberezin-nshl** [comment]
@cindy-peng JFYI, I've just rebased and force pushed again

**Comment by kberezin-nshl** [comment]
@cindy-peng could you please incorporate the changes in a way the most convenient for you, as fast as you can? 
Feel free to cherry-pick the commit and decline this PR, or to just copy-paste the changes, or whatever way fits best for you.

We just want this in as it is holding us off big time.

**Comment by cindy-peng** [comment]
> @cindy-peng could you please incorporate the changes in a way the most convenient for you, as fast as you can? Feel free to cherry-pick the commit and decline this PR, or to just copy-paste the changes, or whatever way fits best for you.
> 
> We just want this in as it is holding us off big time.

Sounds good @kberezin-nshl. If you don't mind, I will cherry-pick this commit and create a new PR. There is some trouble getting the post processing tests passed from the external contributors. Sorry about the delay. I will get the PR out early next week.

**Comment by cindy-peng** [comment]
Close this PR due to no response was returned from OwlBot Post Processor. Here is the new PR: https://github.com/googleapis/java-logging/pull/1574/files

**Pull request closed without merging**